server/initializers/checker.ts

```diff
@@ -56,13 +56,34 @@ function checkMissedConfig () {
     'instance.default_nsfw_policy', 'instance.robots',
     'services.twitter.username', 'services.twitter.whitelisted'
   ]
+  const requiredAlternatives = [
+    [ // set
+      ['redis.hostname', 'redis.port'], // alternative
+      ['redis.socket']
+    ]
+  ]
   const miss: string[] = []
 
   for (const key of required) {
     if (!config.has(key)) {
       miss.push(key)
     }
   }
+  for (const set of requiredAlternatives) {
+    let hasAlternative = false
+    for (const alternative of set) {
+      for (const key of alternative) {
+        if (!config.has(key)) {
+          break
+        }
+        hasAlternative = true
+      }
+      if (hasAlternative) break
+    }
+    if (!hasAlternative) {
+      set[0].forEach((key) => miss.push(key))
+    }
+  }
```

Comment:
This always pushes to `miss` when specifying `redis.socket` alone, and I think it would not add to `miss` if only `redis.port` is defined as well.

Might I suggest:

```typescript
let missingAlternatives = requiredAlternatives.filter(
  set => !set.find(alternative => !alternative.find(key => !config.has(key)))
)

missingAlternatives
  .forEach(set => set[0].forEach(key => miss.push(key))
```